### PR TITLE
feat: Add codecov to CI/CD and update coverage in README.md

### LINKED ISSUE #26: Add codecov to CI/CD and update coverage in README.md
Replace the current tool (basic calculator) with codecov in github actions workflow and show coverage in README.md.

Suggested Gh Action to use: https://github.com/codecov/codecov-action

If you'd like to work on this, please comment "picking this up" below, and I'll assign the issue to you.

Please take a look at our [CONTRIBUTING.md](https://github.com/gopher-utils/go-collections/blob/main/CONTRIBUTING.md) before you start working on this.

### PR DESCRIPTION
**Pull request by JoseAngel1196**
### Description

In this PR, we're adding codecov and updating the README.md to add a new badge to show the coverage of the project.

### Related Issues

Closes #26

### CODE REVIEW

**Comment by JoseAngel1196** [comment]
Hey @aswath-s-tw this is still a draft PR but can you approve the workflow so I can test?

**Review by adarsh-a-tw — APPROVED** [review]

**Comment by adarsh-a-tw** [comment]
Can you add this markdown as badge in README.md? 

`[codecov](https://codecov.io/gh/gopher-utils/go-collections)`

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        os:
+          [
+            "windows-2019",
+            "windows-2022",
+            "ubuntu-20.04",
+            "ubuntu-22.04",
+            "macos-11",
+            "macos-12",
+            "macos-13",
+          ]
```
Comment (JoseAngel1196) [inline .github/workflows/tests.yml]: Making sure the repository works for every OS.

**Comment by JoseAngel1196** [comment]
> Can you add this markdown as badge in README.md?
> 
> `[codecov](https://codecov.io/gh/gopher-utils/go-collections)`

Done! This is ready

**Comment by JoseAngel1196** [comment]
Hey @adarsh-a-tw can you check this PR again?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (adarsh-a-tw) [reply .github/workflows/tests.yml]: @JoseAngel1196 Lets only check for the latest OS.
```suggestion
          [
            "windows-latest",
            "ubuntu-latest",
            "macos-latest",
          ]
```

Comment (JoseAngel1196) [reply .github/workflows/tests.yml]: Done!

**Review by adarsh-a-tw — APPROVED** [review]

**Comment by JoseAngel1196** [comment]
Looks like I can't merge the PR, feel free to merge it!

**Comment by adarsh-a-tw** [comment]
@JoseAngel1196 Thanks for your contribution!

**Pull request merged**